Benqi-Agents/Large-Delegate-Votes-Change/src/agent.ts

```diff
@@ -0,0 +1,55 @@
+import { BigNumber } from "ethers";
+import {
+  Finding,
+  FindingSeverity,
+  FindingType,
+  HandleTransaction,
+  LogDescription,
+  TransactionEvent,
+} from "forta-agent";
+import { DELEGATE_VOTES_CHANGED_ABI, PERCENTAGE_THRESHOLD, QI_ADDRESS } from "./constants";
+
+const isLarge = (previousBalance: BigNumber, newBalance: BigNumber, percentageThreshold: BigNumber): boolean => {
+  const absoluteThreshold = previousBalance.mul(BigNumber.from(100).add(percentageThreshold)).div(100);
+  return newBalance.gte(absoluteThreshold);
+};
+
+export const provideHandleTransaction = (
+  contractAddress: string,
+  percentageThreshold: BigNumber
+): HandleTransaction => {
+  const description = `There was a >= ${percentageThreshold.toString()}% increase in delegate votes based on the previous amount`;
+
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+
+    txEvent.filterLog(DELEGATE_VOTES_CHANGED_ABI, contractAddress).forEach((log: LogDescription) => {
+      if (!isLarge(log.args.previousBalance, log.args.newBalance, percentageThreshold)) {
+        return;
+      }
+
```

Comment:
I think this return can be avoided by changing the if statement condition to `if (isLarge.. ) findings.push(...) `

Reply:
Sure, I did it this way because I thought readability was better than having one more level of nesting, but it's honestly not that big of a change in this case, so I'm changing this.